integration/suite_test.go

```diff
@@ -0,0 +1,191 @@
+	"github.com/keikumata/azure-pub-sub/publisher"
+	"github.com/stretchr/testify/suite"
+)
+
+type serviceBusSuite struct {
+	test.BaseSuite
+	Prefix              string
+	TopicName           string
+	Publisher           publisher.Publisher
+	Listener            listener.Listener
+	publisherAuthOption publisher.ManagementOption
+	listenerAuthOption  listener.ManagementOption
+}
+
+type retryLaterEvent struct {
+	ID    int    `json:"id"`
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type testEvent struct {
+	ID    int    `json:"id"`
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+const (
+	defaultTimeout        = 60 * time.Second
+	testTopicName         = "testTopic"
+	serviceBusResourceURI = "https://servicebus.azure.net/"
+)
+
+func TestConnectionString(t *testing.T) {
+	t.Parallel()
+	connectionStringSuite := &serviceBusSuite{
+		Prefix:              "conn-",
+		listenerAuthOption:  withEnvConnectionString(),
+		publisherAuthOption: withEnvPublisherConnectionString(),
+	}
+	suite.Run(t, connectionStringSuite)
+}
+
+func TestClientId(t *testing.T) {
+	t.Parallel()
+	clientIdSuite := &serviceBusSuite{
+		Prefix:              "cid-",
+		listenerAuthOption:  withEnvManagedIdentityClientID(),
+		publisherAuthOption: withPublisherEnvManagedIdentityClientID(),
+	}
+	suite.Run(t, clientIdSuite)
+}
+
+func TestResourceID(t *testing.T) {
+	t.Parallel()
+	resourceIdSuite := &serviceBusSuite{
+		Prefix:              "rid-",
+		listenerAuthOption:  withEnvManagedIdentityResourceID(),
+		publisherAuthOption: withPublisherEnvManagedIdentityResourceID(),
+	}
+	suite.Run(t, resourceIdSuite)
+}
+
+func withEnvConnectionString() listener.ManagementOption {
+	connStr := os.Getenv("SERVICEBUS_CONNECTION_STRING") // `Endpoint=sb://XXXX.servicebus.windows.net/;SharedAccessKeyName=XXXX;SharedAccessKey=XXXX`
+	if connStr == "" {
+		panic("environment variable SERVICEBUS_CONNECTION_STRING was not set")
+	}
+	return listener.WithConnectionString(connStr)
+}
+
+func withEnvPublisherConnectionString() publisher.ManagementOption {
+	connStr := os.Getenv("SERVICEBUS_CONNECTION_STRING") // `Endpoint=sb://XXXX.servicebus.windows.net/;SharedAccessKeyName=XXXX;SharedAccessKey=XXXX`
+	if connStr == "" {
+		panic("environment variable SERVICEBUS_CONNECTION_STRING was not set")
+	}
+
+	return publisher.WithConnectionString(connStr)
+}
+
+func withEnvManagedIdentityClientID() listener.ManagementOption {
+	serviceBusNamespaceName := os.Getenv("SERVICEBUS_NAMESPACE_NAME") // `Endpoint=sb://XXXX.servicebus.windows.net/;SharedAccessKeyName=XXXX;SharedAccessKey=XXXX`
+	if serviceBusNamespaceName == "" {
+		panic("environment variable SERVICEBUS_NAMESPACE_NAME was not set")
+	}
+
+	// if managedIdentityClientID is empty then library will assume system assigned managed identity
+	managedIdentityClientID := os.Getenv("MANAGED_IDENTITY_CLIENT_ID")
+	if managedIdentityClientID == "" {
+		panic("environment variable MANAGED_IDENTITY_CLIENT_ID was not set")
+	}
```

Comment:
looks like we don't allow system assigned MSI during the test?

Reply:
Ah that's right. I can remove that check

Reply:
fixed